cpp/tests/test_utils.h

```diff
@@ -62,6 +62,23 @@ extern "C" {
 }
 #endif
 
+#define NCCLCHECK(cmd) do {                                   \
+    ncclResult_t nccl_status = cmd;                           \
+    if (nccl_status!= ncclSuccess) {                          \
+      printf("NCCL failure %s:%d '%s'\n",                     \
+          __FILE__,__LINE__,ncclGetErrorString(nccl_status)); \
+    }                                                 \
+  } while(0)
```

Comment:
Can you explain (preferably in a code comment for others) why the 1-iteration do-while loop is needed? Is this to force a new scope?

Reply:
like the printf in the comment above, these are artifacts from debugging the prototype, I need to update it for gtest.

Reply:
I think the original source for these macros was NCCL doc https://docs.nvidia.com/deeplearning/sdk/nccl-developer-guide/docs/examples.html